### DISCUSSION TITLE
Why isn't `include_cached` tag used in `_layouts/single.html` (and other layouts)?

### DISCUSSION THREAD

**Discussion by jlconlin** [mmistakes/minimal-mistakes#3794] [Q&A] [answered]
I recently learned about the `jekyll-include-cache` gem, which provides the `include_cached` liquid tag. I would *really* like to speed up the building/serving of my site. 

The `single.html` file accounts for the majority of time spent rending my site (followed closely by  `default.html` and `head.html`). This makes sense because I have a few thousand posts that all use the `single` layout.

I looked at `_layouts/single.html` and saw that it did *not* use the `include_cached` tag, opting instead to use the standard `include` tag. Since this layout seems to be so important, **why isn't `include_cached` used more liberally?**

**Comment by iBug** [comment, marked as answer]
Because `include_cached` reuses previously rendered results based only on the `{% include %}` directive. All includes in `single.html` refers to the current page via either `{{ page.* }}` or `{{ post.* }}`. If `include_cached` is used, all posts would contain wrong information except the first post processed.

**Reply by jlconlin** [reply]
That makes sense. Thanks for the information
